enclave-runtime/src/lib.rs

```diff
@@ -135,6 +147,18 @@ pub extern "C" fn test_main_entrance() -> size_t {
 pub const CERTEXPIRYDAYS: i64 = 90i64;
 
 pub type Hash = sp_core::H256;
+pub type AuthorityPair = sp_core::ed25519::Pair;
+
+lazy_static! {
+	/// the enclave's parentchain nonce
+	///
+	/// This should be abstracted away better in the future. Currently, this only exists
+	/// because we produce sidechain blocks faster that parentchain chain blocks. So for now this
+	/// design suffices. Later, we also need to sync across parallel ecalls that might both result
+	/// in parentchain xt's. Then we should probably think about how we want to abstract the global
+	/// nonce.
+	static ref NONCE: SgxRwLock<u32> = SgxRwLock::new(Default::default());
+}
```

Comment:
Not a big fan of this. I would rather we have an additional layer of access to a global variable like this (like I've done for all the previous global variables). So you can at least pass in that abstracted access layer and not have access to the global variable directly. It creates a dependency nightmare and makes code less testable.

Reply:
we could consider each shard having its own enclave key. I don't really like that but it might be less complex than coordinating parallel running shards for the nonce

Reply:
I am against a per-shard key because it would not completely remove the need of a per-shard-global nonce. I think we can solve that quite well with the extrinsic factory proposed here: #429